### PR TITLE
[Multi Hub] Tab Calander- Nodejs Sample

### PR DESCRIPTION
**Pull request by Harikrishnan-MSFT**

### CODE REVIEW

**Comment by Harikrishnan-MSFT** [comment]
Build Pipeline - 
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5320066977
image

File: samples/tab-calendar/ts/Manifest/manifest.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.16/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.16",
+  "version": "1.0.0",
+  "id": "{{Microsoft-App-Id}}",
+  "packageName": "com.teams.TabCalendar",
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/Manifest/manifest.json:6]: TabCalendar => tabcalendar

File: samples/tab-calendar/ts/Manifest/manifest.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "icons": {
+    "color": "color.png",
+    "outline": "outline.png"
+  },
+  "name": {
+    "short": "Tab Calendar",
+    "full": "Tab Calendar"
+  },
+  "description": {
+    "short": "Short description for Tab Calendar",
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/Manifest/manifest.json:22]: Interact with Calendar in Outlook from Personal Tab

File: samples/tab-calendar/ts/Manifest/manifest.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "full": "Full description of Interact with the user's calendar, including opening calendar items and composing meetings."
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/Manifest/manifest.json:23]: Interact with the user's calendar, including opening calendar items and composing meetings from Personal Tab in Outlook.

File: samples/tab-calendar/ts/Manifest/manifest.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  "accentColor": "#5558AF",
+  "staticTabs": [
+    {
+      "contentUrl": "https://{{domain-name}}/tab",
+      "websiteUrl": "https://{{domain-name}}/tab",
+      "entityId": "com.contoso.TabCalendar.help",
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/Manifest/manifest.json:30]: com.contoso.TabCalendar.help => com.contoso.tabcalendar.help

File: samples/tab-calendar/ts/README.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
+---
+page_type: sample
+description: This sample shows tab calendar in Outlook apps.
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/README.md:3]: This sample shows how to interact with Calendar in Outlook from Personal Tab

File: samples/tab-calendar/ts/README.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+- office-365
+languages:
+- typescript
+extensions:
+ contentType: samples
+ createdDate: "20/06/2023 01:38:27 PM"
+urlFragment: officedev-microsoft-teams-samples-tab-calendar-ts
+---
+
+# Tab Calendar in Outlook
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/README.md:16]: Interact with Calendar in Outlook from Personal Tab

File: samples/tab-calendar/ts/README.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+Interact with the user's calendar, including opening calendar items and composing meetings.
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/README.md:18]: Interact with the user's calendar, including opening calendar items and composing meetings from Personal Tab in Outlook.

File: samples/tab-calendar/ts/README.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+ ## Included Features
+* Tabs
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/README.md:21]: Add Outlook Calendar as well

File: samples/tab-calendar/ts/README.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+**Click ComposeMeeting:**
+
+![ClickComposeMeeting](Images/ClickComposeMeeting.png)
+
+**Click View:**
+
+![ClickView](Images/ClickView.png)
+
+## Further Reading.
+[Calendar Module](https://learn.microsoft.com/en-us/javascript/api/@microsoft/teams-js/calendar?view=msteams-client-js-latest)
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/README.md:159]: Add this link as well:
https://learn.microsoft.com/en-us/microsoftteams/platform/m365-apps/teamsjs-support-m365#calendar

File: samples/tab-calendar/ts/api-server/server.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+app.get('/getGraphAccessToken', async (req,res) => {
+
+    const msalClient = new msal.ConfidentialClientApplication({
+        auth: {
+            clientId: clientId,
+            clientSecret: clientSecret
+        }
+    });
+    let tenantId = jwt_decode(req.query.ssoToken)['tid']; //Get the tenant ID from the decoded toke
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/api-server/server.js:29]: One line space before it.

File: samples/tab-calendar/ts/api-server/server.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+    msalClient.acquireTokenOnBehalfOf({
+        authority: `https://login.microsoftonline.com/${tenantId}`,
+        oboAssertion: req.query.ssoToken,
+        scopes: graphScopes,
+        skipCache: true
+      })
+      .then( async (result) => {     
+                let graphPhotoEndpoint = `https://graph.microsoft.com/v1.0/users/${req.query.upn}/events?$select=subject,body,bodyPreview,organizer,attendees,start,end,location`;
+                let graphRequestParams = {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/api-server/server.js:39]: One line space before it.

File: samples/tab-calendar/ts/api-server/server.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+                    method: 'GET',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        "authorization": "bearer " + result.accessToken
+                    }
+                }
+                let response = await fetch(graphPhotoEndpoint,graphRequestParams).catch(this.unhandledFetchError);
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/api-server/server.js:46]: One line space before it.

File: samples/tab-calendar/ts/api-server/server.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+                if (response.ok) {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/api-server/server.js:47]: One line space before it.

File: samples/tab-calendar/ts/assets/sample.json [hunk 14]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-tab-calendar-ts",
+    "source": "officeDev",
+    "title": "Tab calendar in Outlook",
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/assets/sample.json:5]: Interact with the calendar in Outlook from personal tab.

File: samples/tab-calendar/ts/assets/sample.json [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+        "value": "OfficeDev"
+      },
+      {
+        "key": "TEAMS-SERVER-LANGUAGE",
+        "value": ""
+      },
+      {
+        "key": "TEAMS-SERVER-PLATFORM",
+        "value": ""
+      },
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/assets/sample.json:28]: Please update these values.

File: samples/tab-calendar/ts/assets/sample.json [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+        "key": "TEAMS-FEATURES",
+        "value": "tab"
+      }
+    ],
+    "thumbnails": [
+      {
+        "type": "image",
+        "order": 100,
+        "url": "https://raw.githubusercontent.com/OfficeDev/Microsoft-Teams-Samples/main/samples/tab-calendar/ts/Images/TabPersonalQuickstartGif.gif",
+        "alt": "Solution UX showing Tab calendar support in Outlook apps."
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/assets/sample.json:39]: Solution UX showing how to interact with Calendar in Outlook from Personal Tab.

File: samples/tab-calendar/ts/public/index.html [hunk 17]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="stylesheet" href="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-core/11.0.0/css/fabric.min.css" />
+    <title>Microsoft Teams Tab</title>
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/public/index.html:7]: Interact with Calendar in Outlook from Personal Tab

File: samples/tab-calendar/ts/src/components/App.css [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+.tdSubject{
+  width: 83%;
+  word-wrap: break-word;
+}
+.mainDivOpen{
+  width: 50%;
+  float: left;
+  margin-top: -4.3%;
+}
+@keyframes App-logo-spin {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/App.css:78]: One line space before it.

File: samples/tab-calendar/ts/src/components/ClosePopup.tsx [hunk 19]
```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import React from 'react';
+import * as microsoftTeams from "@microsoft/teams-js";
+import * as msal from "@azure/msal-browser";
+
+/**
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/ClosePopup.tsx:8]: Add single line comment

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 20]
```diff
@@ -0,0 +1,380 @@
...
+/**
+ * This tab component renders the main tab content
+ * of your app.
+ */
+
+export interface IAllItemsDetail {
+  id: string;
+  subject: string;
+}
+export interface ITabProps {}
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:16]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 21]
```diff
@@ -0,0 +1,380 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import React from "react";
+import "./App.css";
+import * as microsoftTeams from "@microsoft/teams-js";
+/**
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:7]: One line space before it.

Also, add single line comment.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 22]
```diff
@@ -0,0 +1,380 @@
...
+  graphAccessToken: string;
+  error: boolean;
+  stateDatetime: string;
+  endDatetime: string;
+  attendees: string;
+  subject: string;
+  content: string;
+  allItems: IAllItemsDetail[];
+}
+class Tab extends React.Component<ITabProps, ITabState> {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:31]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 23]
```diff
@@ -0,0 +1,380 @@
...
+
+    this.consentSuccess = this.consentSuccess.bind(this);
+    this.consentFailure = this.consentFailure.bind(this);
+    this.unhandledFetchError = this.unhandledFetchError.bind(this);
+    this.callGraphFromClient = this.callGraphFromClient.bind(this);
+    this.showConsentDialog = this.showConsentDialog.bind(this);
+  }
+
+  //React lifecycle method that gets called once a component has finished mounting
+  //Learn more: https://reactjs.org/docs/react-component.html#componentdidmount
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:65]: // React lifecycle method that gets called once a component has finished mounting
// Learn more: https://reactjs.org/docs/react-component.html#componentdidmount

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 24]
```diff
@@ -0,0 +1,380 @@
...
+        });
+    });
+  }
+
+  //Exchange the SSO access token for a Graph access token
+  //Learn more: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow
+  exchangeClientTokenForServerToken = async (token: string) => {
+    let serverURL = `${process.env.REACT_APP_BASE_URL}/getGraphAccessToken?ssoToken=${token}&upn=${this.state.context?.user?.userPrincipalName}`;
+    let response = await fetch(serverURL).catch(this.unhandledFetchError); //This calls getGraphAccessToken route in /api-server/app.js
+    if (response) {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:93]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 25]
```diff
@@ -0,0 +1,380 @@
...
+      let data = await response.json().catch(this.unhandledFetchError);
+      if (!response.ok && data.error === "consent_required") {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:95]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 26]
```diff
@@ -0,0 +1,380 @@
...
+        console.error(data);
+        this.setState({ error: true });
+      } else {
+        this.setState({ allItems: data.value });
+      }
+    }
+  };
+
+  //Show a popup dialogue prompting the user to consent to the required API permissions. This opens ConsentPopup.js.
+  //Learn more: https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/auth-tab-aad#initiate-authentication-flow
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:111]: // Show a popup dialogue prompting the user to consent to the required API permissions. This opens ConsentPopup.js.
// Learn more: https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/auth-tab-aad#initiate-authentication-flow

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 27]
```diff
@@ -0,0 +1,380 @@
...
+      })
+      .then((result: string) => {
+        this.consentSuccess(result ?? "");
+      })
+      .catch((reason: string) => {
+        this.consentFailure(reason ?? "");
+      });
+  }
+
+  //Callback function for a successful authorization
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:127]: // Callback function for a successful authorization.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 28]
```diff
@@ -0,0 +1,380 @@
...
+      consentProvided: true,
+    });
+  }
+
+  consentFailure(reason: string) {
+    console.error("Consent failed: ", reason);
+    this.setState({ error: true });
+  }
+
+  //React lifecycle method that gets called after a component's state or props updates
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:141]: // React lifecycle method that gets called after a component's state or props updates
  // Learn more: https://reactjs.org/docs/react-component.html#componentdidupdate

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 29]
```diff
@@ -0,0 +1,380 @@
...
+    let upn = this.state.context?.user?.userPrincipalName;
+    let graphEndpoint = `https://graph.microsoft.com/v1.0/users/${upn}/events?$select=subject,body,bodyPreview,organizer,attendees,start,end,location`;
+    let graphRequestParams = {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        authorization: "bearer " + this.state.graphAccessToken,
+      },
+    };
+    let response = await fetch(graphEndpoint, graphRequestParams).catch(
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:165]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 30]
```diff
@@ -0,0 +1,380 @@
...
+      this.unhandledFetchError
+    );
+    if (response) {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:168]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 31]
```diff
@@ -0,0 +1,380 @@
...
+      let data = await response.json().catch(this.unhandledFetchError);
+      if (response.ok) {
+        this.setState({ allItems: data.value });
+      } else {
+        console.error("ERROR: ", response);
+        this.setState({ error: true });
+      }
+    }
+  };
+  //Generic error handler ( avoids having to do async fetch in try/catch block )
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:178]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 32]
```diff
@@ -0,0 +1,380 @@
...
+  }
+  attendeesHandleChange(ev: any) {
+    const attendeesData = ev.target.value;
+    this.setState({ attendees: attendeesData });
+  }
+  contentHandleChange(ev: any) {
+    const setContentData = ev.target.value;
+    this.setState({ content: setContentData });
+  }
+  openCalendar(itemIds: any) {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:206]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 33]
```diff
@@ -0,0 +1,380 @@
...
+      this.state.stateDatetime !== undefined &&
+      this.state.stateDatetime !== null &&
+      this.state.stateDatetime !== "" &&
+      this.state.endDatetime !== undefined &&
+      this.state.endDatetime !== null &&
+      this.state.endDatetime !== ""
+    ) {
+      isValidation = true;
+    }
+    if (isValidation == true) {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:238]: One line space before it.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 34]
```diff
@@ -0,0 +1,380 @@
...
+    if (microsoftTeams.calendar.isSupported()) {
+      microsoftTeams.calendar.openCalendarItem(OpenCalendarItemParams);
+    } else {
+      alert("Calendar is not supported!!!");
+    }
+  }
+
+  composeMeeting() {
+    var isValidation = false;
+    if (
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:219]: Alignment.

File: samples/tab-calendar/ts/src/components/Tab.tsx [hunk 35]
```diff
@@ -0,0 +1,380 @@
...
+    }
+    if (isValidation == true) {
+      var ComposeMeetingParams = {
+        attendees: [this.state.attendees],
+        startTime: this.state.stateDatetime, // MM/DD/YYYY HH:MM:SS format
+        endTime: this.state.endDatetime, // MM/DD/YYYY HH:MM:SS format
+        subject: this.state.subject,
+        content: this.state.content,
+      };
+      if (microsoftTeams.calendar.isSupported()) {
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/Tab.tsx:246]: One line space before it.

File: samples/tab-calendar/ts/src/components/TabCalendarMain.tsx [hunk 36]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * This component is used to display the required
+ * privacy statement which can be found in a link in the
+ * about tab.
+ */
+class TabCalendar extends React.Component {
+    render() {
+      return (
+        <div>
+          <h1>Welcome to interacting with the user's calendar, including opening calendar items and composing meetings in calendar apps</h1>
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/src/components/TabCalendarMain.tsx:17]: Welcome to interacting with the user's calendar, including opening calendar items and composing meetings from Personal Tab app in Outlook.

File: README.md [hunk 37]
```diff
@@ -53,7 +53,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |20| Tab app monetization | This sample shows how to open purchase dialog and trigger purchase flow using teams-js sdk. ||[View][tab-app-monetization#nodejs]|
 |21| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
 |22| App Caching in Meetings | This sample shows how to improve the subsequent loading time of an App. |||[View][app-cache-meetings#nodejs]|
-|23| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
```
[the remark this answers is not in the document]
Comment (ChetanSharma-msft) [reply README.md:56]: Removed duplicate entry.

File: .github/workflows/build-complete-samples.yml [hunk 38]
```diff
@@ -898,6 +898,14 @@ jobs:
             - project_path: 'samples/meetings-app-icon-badging/nodejs'
               name: 'meetings-app-icon-badging'
               version: '18.x'
+            
+            - project_path: 'samples/tab-calendar/ts/api-server'
+              name: 'tab-calendar-api-server'
```
Comment (ChetanSharma-msft) [inline .github/workflows/build-complete-samples.yml:903]: Need to create deploy pipeline.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

File: README.md [hunk 39]
```diff
@@ -53,7 +53,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
...
+|23| Tab Calendar in Outlook | Interact with the user's calendar, including opening calendar items and composing meetings.||[View][tab-calendar#ts]|
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply README.md:56]: Fixed

File: samples/tab-calendar/ts/Manifest/manifest.json (hunk 1 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/Manifest/manifest.json:6]: Fixed

File: samples/tab-calendar/ts/Manifest/manifest.json (hunk 2 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/Manifest/manifest.json:22]: Fixed

File: samples/tab-calendar/ts/Manifest/manifest.json (hunk 3 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/Manifest/manifest.json:23]: Fixed

File: samples/tab-calendar/ts/Manifest/manifest.json (hunk 4 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/Manifest/manifest.json:30]: Fixed

File: samples/tab-calendar/ts/README.md (hunk 5 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/README.md:3]: Fixed

File: samples/tab-calendar/ts/README.md (hunk 6 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/README.md:16]: Fixed

File: samples/tab-calendar/ts/README.md (hunk 7 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/README.md:18]: Fixed

File: samples/tab-calendar/ts/README.md (hunk 8 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/README.md:21]: Fixed

File: samples/tab-calendar/ts/README.md (hunk 9 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/README.md:159]: Fixed

File: samples/tab-calendar/ts/api-server/server.js (hunk 10 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/api-server/server.js:29]: Fixed

File: samples/tab-calendar/ts/api-server/server.js (hunk 11 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/api-server/server.js:39]: Fixed

File: samples/tab-calendar/ts/api-server/server.js (hunk 12 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/api-server/server.js:46]: Fixed

File: samples/tab-calendar/ts/api-server/server.js (hunk 13 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/api-server/server.js:47]: Fixed

File: samples/tab-calendar/ts/assets/sample.json (hunk 14 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/assets/sample.json:5]: Fixed

File: samples/tab-calendar/ts/assets/sample.json [hunk 40]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-tab-calendar-ts",
+    "source": "officeDev",
+    "title": "Tab calendar in Outlook",
+    "shortDescription": "Create React App personal tab in TypeScript",
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/tab-calendar/ts",
+    "longDescription": [
+      "This sample app shows interact with the user's calendar, including opening calendar items and composing meetings."
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/assets/sample.json:9]: Fixed

File: samples/tab-calendar/ts/assets/sample.json [hunk 41]
```diff
@@ -0,0 +1,68 @@
...
+    ],
+    "creationDateTime": "2023-06-19",
+    "updateDateTime": "2023-06-10",
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/assets/sample.json:12]: Fixed

File: samples/tab-calendar/ts/assets/sample.json [hunk 42]
```diff
@@ -0,0 +1,68 @@
...
+        "key": "TEAMS-SERVER-LANGUAGE",
+        "value": ""
+      },
+      {
+        "key": "TEAMS-SERVER-PLATFORM",
+        "value": ""
+      },
+      {
+        "key": "TEAMS-FEATURES",
+        "value": "tab"
```
[the remark this answers is not in the document]
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/assets/sample.json:31]: Fixed

File: samples/tab-calendar/ts/assets/sample.json (hunk 15 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/assets/sample.json:28]: Fixed

File: samples/tab-calendar/ts/assets/sample.json (hunk 16 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/assets/sample.json:39]: Fixed

File: samples/tab-calendar/ts/public/index.html (hunk 17 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/public/index.html:7]: Fixed

File: samples/tab-calendar/ts/src/components/App.css (hunk 18 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/App.css:78]: Fixed

File: samples/tab-calendar/ts/src/components/ClosePopup.tsx (hunk 19 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/ClosePopup.tsx:8]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 20 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:16]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 21 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:7]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 22 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:31]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 23 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:65]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 24 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:93]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 25 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:95]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 26 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:111]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 27 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:127]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 28 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:141]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 29 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:165]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 30 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:168]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 31 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:178]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 33 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:238]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 34 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:219]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 35 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:246]: Fixed

File: samples/tab-calendar/ts/src/components/TabCalendarMain.tsx (hunk 36 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/TabCalendarMain.tsx:17]: Fixed

File: samples/tab-calendar/ts/src/components/Tab.tsx (hunk 32 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/src/components/Tab.tsx:209]: Fixed

File: README.md [hunk 43]
```diff
@@ -53,7 +53,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |20| Tab app monetization | This sample shows how to open purchase dialog and trigger purchase flow using teams-js sdk. ||[View][tab-app-monetization#nodejs]|
 |21| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
 |22| App Caching in Meetings | This sample shows how to improve the subsequent loading time of an App. |||[View][app-cache-meetings#nodejs]|
-|23| Personal Tab with Nav-Bar menu | Add multiple actions to the upper right in Nav-Bar and build an overflow menu for extra actions in an app. ||[View][tab-navbar-menu#ts]|
+|23| Tab calendar in Outlook | Interact with the user's calendar, including opening calendar items and composing meetings from personal tab in Outlook.||[View][tab-calendar#ts]|
```
Comment (ChetanSharma-msft) [inline README.md:56]: This sample shows how to interact with the user's calendar, including opening calendar items and composing meetings from personal tab in Outlook.

File: samples/tab-calendar/ts/Manifest/manifest.json [hunk 44]
```diff
@@ -0,0 +1,48 @@
...
+    "color": "color.png",
+    "outline": "outline.png"
+  },
+  "name": {
+    "short": "Tab Calendar",
+    "full": "Tab Calendar"
+  },
+  "description": {
+    "short": "Interact with calendar in Outlook from personal tab",
+    "full": "Interact with the user's calendar, including opening calendar items and composing meetings from personal tab in Outlook."
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/Manifest/manifest.json:23]: Interact with the user's calendar, including opening calendar items and composing meetings from personal tab in Outlook.

File: samples/tab-calendar/ts/public/index.html [hunk 45]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="stylesheet" href="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-core/11.0.0/css/fabric.min.css" />
+    <title>Interact with calendar in Outlook from personal tab</title>
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/public/index.html:7]: This sample app shows how to interact with the user's calendar, including opening calendar items and composing meetings from personal tab in Outlook.

**Review by ChetanSharma-msft — APPROVED** [review]
Added few minor comments.

File: README.md (hunk 43 above)
Comment (Harikrishnan-MSFT) [reply README.md:56]: Fixed

File: samples/tab-calendar/ts/Manifest/manifest.json (hunk 44 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/Manifest/manifest.json:23]: Fixed

File: samples/tab-calendar/ts/public/index.html (hunk 45 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/public/index.html:7]: Fixed

File: samples/tab-calendar/ts/README.md [hunk 46]
```diff
@@ -0,0 +1,164 @@
...
+    npm install
+    ```
+
+    ```bash
+    npm start
+    ```
+- The client will start running on 3978 port
+
+4. Setup Manifest for Teams
+
```
Comment (ChetanSharma-msft) [inline samples/tab-calendar/ts/README.md:126]: Please verify readme file:
image

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added one review comment.

File: samples/tab-calendar/ts/README.md (hunk 46 above)
Comment (Harikrishnan-MSFT) [reply samples/tab-calendar/ts/README.md:126]: Fixed

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Comment by Harikrishnan-MSFT** [comment]
**Build Pipeline -**
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/5419679210

image

**Comment by ChetanSharma-msft** [comment]
Duplicate of: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/852

**Pull request closed without merging**